src/geometry/euclidean/inversion.lean

```diff
@@ -0,0 +1,91 @@
+/-
+Copyright (c) 2022 Yury G. Kudryashov. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Yury G. Kudryashov
+-/
+import geometry.euclidean.basic
+
+/-!
+# Inversion in an affine space
+
+In this file we define inversion in a sphere in an affine space. This map sends each point `x` to
+the point `y` such that `y -ᵥ c = (R / dist x c) ^ 2 • (x -ᵥ c)`, where `c` and `R` are the center
+and the radius the sphere.
+
+In many applications, it is convenient to assume that the inversions swaps the center and the point
+at infinity. In order to stay in the original affine space, we define the map so that it sends
+center to itself.
+
+Currently, we prove only a few basic lemmas needed to prove Ptolemy's inequality, see
+`euclidean_geometry.mul_dist_le_mul_dist_add_mul_dist`.
+-/
+
+noncomputable theory
+open metric real
+
+namespace euclidean_geometry
+
+variables {V P : Type*} [inner_product_space ℝ V] [metric_space P] [normed_add_torsor V P]
+  {a b c d x y z : P} {R : ℝ}
+
+include V
+
+/-- Inversion in a sphere in an affine space. This map sends each point `x` to the point `y` such
+that `y -ᵥ c = (R / dist x c) ^ 2 • (x -ᵥ c)`, where `c` and `R` are the center and the radius the
+sphere. -/
+def inversion (c : P) (R : ℝ) (x : P) := (R / dist x c) ^ 2 • (x -ᵥ c) +ᵥ c
+
+@[simp] lemma inversion_self (c : P) (R : ℝ) : inversion c R c = c := by simp [inversion]
+
```

Comment:
Can you add the lemma that `inversion c R (inversion c R x) = x` (and the statement using `function.involutive` if it's true with no extra hypotheses)?